messaging/src/main/java/org/axonframework/eventhandling/pooled/Coordinator.java

```diff
@@ -649,12 +679,32 @@ public void run() {
                 return;
             }
 
+            // Abort WorkPackages for which releaseUntil() has been invoked
             workPackages.entrySet().stream()
                         .filter(entry -> isSegmentBlockedFromClaim(entry.getKey()))
                         .map(Map.Entry::getValue)
                         .forEach(workPackage -> abortWorkPackage(workPackage, null));
 
+            if (coordinatorExtendsClaims) {
+                logger.debug("Processor [{}] extending all claims of active Work Packages.", name);
+                // Extend the claims of each work package, relieving this effort from the work package as an optimization
+                workPackages.values()
+                            .stream()
+                            .filter(workPackage -> !workPackage.isAbortTriggered())
+                            .forEach(workPackage -> {
+                                try {
+                                    workPackage.extendClaim();
```

Comment:
This would happen on every coordinator pass, which is quite a lot. We could add a mechanism (preferably in the `extendClaim` method) to avoid excessive extensions by looking at the time at which the last extension (or update of the token, which also extends the claim) was executed.

Reply:
Resolved this by adjusting the `extendClaim` method into `extendClaimIfThresholdIsMet`. Doing so, the `Coordinator` also partakes in the `lastClaimExtension` timer.